### ISSUE TITLE
Segmentation fault when indexing large axes

### ISSUE THREAD

**Issue by neishm** [pygeode/pygeode#107]
Indexing into a very large axis can trigger a segmentation fault.  On my test system, the following code can trigger the bug:
```python
from pygeode.axis import NamedAxis
from pygeode.var import Var

n = 1000000
obsnum = NamedAxis(values=range(n), name='obsnum')
data = Var([obsnum], values=[100.]*n, name='data')

val = data[0]
```

This appears to be happening in the **map_to** function in **tools.c**.  In there, a copy of the axes is placed onto the stack, which will cause problems if the axis is larger than the stack size.
https://github.com/pygeode/pygeode/blob/7de59afeef17c9bfbbaac4a895b1b9077b950df4/pygeode/tools.c#L97-L104

I believe the solution would be to change the **map_to** code so that **a**, **b**, **a_ind**, and **b_ind** are allocated on the heap (using malloc / free).

**Comment by aph42** [comment]
Yep - it segfaults on my machine too. Seems like the right solution. Should we impose some kind of maximum axis length?

**Comment by neishm** [comment]
There are legitimate netCDF files which exceed the limit, such as 1D station data with high temporal frequency.  The original crash came from an ObsPack dataset that a co-worker was trying to process with PyGeode.

I just created a pull request which I think resolves any potential overflows in the C code, except maybe the EOF and SVD routines, which could potentially crash if (num_eofs)**2 exceeds the stack size.  Realistically I don't think anyone would be computing that many leading EOFs.
